react-native/react/menubar/index.js

```diff
@@ -40,12 +41,20 @@ class Menubar extends Component {
     const onMenubarShow = () => {
       setImmediate(() => {
         this.checkForFolders()
+
+        engine.listenOnConnect('menubar', () => {
```

Comment:
I mean that you don't need to do the first call to check folders since if the engine is connected that callback will happen anyways